src/server/string_family.cc

```diff
@@ -532,9 +540,18 @@ struct GetReplies {
 }  // namespace
 
 StringValue StringValue::Read(DbIndex dbid, string_view key, const PrimeValue& pv,
-                              EngineShard* es) {
-  return pv.IsExternal() ? StringValue{es->tiered_storage()->Read(dbid, key, pv)}
-                         : StringValue(GetString(pv));
+                              DbSlice* db_slice) {
+  auto* ts = db_slice->shard_owner()->tiered_storage();
+  if (pv.IsExternal()) {
+    db_slice->IncRamMisses();
+    return StringValue{ts->Read(dbid, key, pv)};
+  }
+
+  db_slice->IncRamHits();
+  if (pv.HasStashPending()) {
+    ts->CancelStash(dbid, key, &pv);
```

Comment:
this is already done by FindInternal

Reply:
it is not, it is handled in PreUpdate

Reply:
so should we move it from there to FindInternal?